contracts/src/instances/savings-usds-l2/SavingsUSDSL2Hyperdrive.sol

```diff
@@ -0,0 +1,100 @@
+///    #;;;;;//////////////////////////////////////////////////////////////;;;;;#
+///  S;;;;;;;;;//////////////////////////////////////////////////////////;;;;;;;;;S
+/// /;;;;;;;;;;;///////////////////////////////////////////////////////;;;;;;;;;;;;\
+/// |||OOOOOOOO||OOOOOOOO=========== __  ___        ===========OOOOOOOO||OOOOOOOO|||
+/// |||OOOOOOOO||OOOOOOOO===========|  \[__ |   \  /===========OOOOOOOO||OOOOOOOO|||
+/// |||OOOOOOOO||OOOOOOOO===========|__/[___|___ \/ ===========OOOOOOOO||OOOOOOOO|||
+/// ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
+/// |||////////000000000000\\\\\\\\|:::::::::::::::|////////00000000000\\\\\\\\\\|||
+/// SSS\\\\\\\\000000000000////////|:::::0x666:::::|\\\\\\\\00000000000//////////SSS
+/// SSS|||||||||||||||||||||||||||||:::::::::::::::||||||||||||||||||||||||||||||SSS
+/// SSSSSSSS|_______________|______________||_______________|______________|SSSSSSSS
+/// SSSSSSSS                                                                SSSSSSSS
+/// SSSSSSSS                                                                SSSSSSSS
+///
+/// @author DELV
+/// @title SavingsUSDSL2Hyperdrive
+/// @notice A Hyperdrive instance that uses a SavingsUSDSL2 vault as the yield source.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+contract SavingsUSDSL2Hyperdrive is Hyperdrive, SavingsUSDSL2Base {
+    using SafeERC20 for ERC20;
+
+    /// @notice Instantiates Hyperdrive with a SavingsUSDSL2 vault as the yield source.
+    /// @param __name The pool's name.
+    /// @param _config The configuration of the Hyperdrive pool.
+    /// @param __adminController The admin controller that will specify the
+    ///        admin parameters for this instance.
+    /// @param _target0 The target0 address.
+    /// @param _target1 The target1 address.
+    /// @param _target2 The target2 address.
+    /// @param _target3 The target3 address.
+    /// @param _target4 The target4 address.
+    /// @param _PSM the PSM contract.
+    constructor(
+        string memory __name,
+        IHyperdrive.PoolConfig memory _config,
+        IHyperdriveAdminController __adminController,
+        address _target0,
+        address _target1,
+        address _target2,
+        address _target3,
+        address _target4,
+        IPSM _PSM
+    )
+        Hyperdrive(
+            __name,
+            _config,
+            __adminController,
+            _target0,
+            _target1,
+            _target2,
+            _target3,
+            _target4
+        )
+        SavingsUSDSL2Base(_PSM)
+    {
+        // Approve the base token with 1 wei. This ensures that all of the
+        // subsequent approvals will be writing to a dirty storage slot.
+        ERC20(address(_config.baseToken)).forceApprove(address(this), 1);
```

Comment:
What’s the point of these approvals? We’re approving this address to spend the tokens, but it should already be able to do that by just calling `transfer` (or using the `SafeERC20.safeTransfer` library function.

Reply:
I'd seen them on other contracts, but didn't quite get why they were there. Removed these now.